random_student_marks_mysql2.php

```diff
@@ -18,104 +16,80 @@
     $teachingRandomSubjects[$i] = $teachingSubjects[$teachingRandomSubject];
 }
 $uniqueStudentNames = array();
-foreach ($studentNames as $key => $value) {
-    if (!in_array($value, $uniqueStudentNames)) {
-        $uniqueStudentNames[] = $value;
+foreach ($studentNames as $names) {
+    if (!in_array($names, $uniqueStudentNames)) {
+        $uniqueStudentNames[] = $names;
     }
 }
 $uniqueTeachingRandomSubjects = array();
-foreach ($teachingRandomSubjects as $key => $value) {
-    if (!in_array($value, $uniqueTeachingRandomSubjects)) {
-        $uniqueTeachingRandomSubjects[] = $value;
+foreach ($teachingRandomSubjects as $key => $subject) {
+    if (!in_array($subject, $uniqueTeachingRandomSubjects)) {
+        $uniqueTeachingRandomSubjects[] = $subject;
     }
 }
-foreach ($uniqueStudentNames as $key => $value) {
+foreach ($uniqueStudentNames as $key => $names) {
     $uniqueStudentName = explode(",", $uniqueStudentNames[$key]);
     $sqlQuerry = "INSERT INTO students (name, surname) VALUES ('$uniqueStudentName[0]', '$uniqueStudentName[1]')";
     mysqli_query($connection, $sqlQuerry);
 }
-foreach ($uniqueTeachingRandomSubjects as $key => $value) {
+foreach ($uniqueTeachingRandomSubjects as $key => $subject) {
     $sqlQuerry = "INSERT INTO teaching_subjects (teachingSubject) VALUES ('$uniqueTeachingRandomSubjects[$key]')";
     mysqli_query($connection, $sqlQuerry);
 }
 $counterStudent = count($uniqueStudentNames);
 $counterSubject = count($uniqueTeachingRandomSubjects);
+define("TIME_STAMP_YEAR_2000", "946684800");
+define("TIME_STAMP_YEAR_2019", "1546300800");
 for ($i = 0; $i < $n; $i++) {
     $studentRandomNameId = rand(1, $counterStudent);
     $randomSubjectId = rand(1, $counterSubject);
-    $dateTimestamp = rand(946684800, 1546300800);
+    $dateTimestamp = rand(TIME_STAMP_YEAR_2000, TIME_STAMP_YEAR_2019);
     $mark = rand(1, 10);
     $date = date("Y-m-d", $dateTimestamp);
     $sqlQuerry = "INSERT INTO marks (idStudent, idSubject, mark, date)
     VALUES ('$studentRandomNameId', '$randomSubjectId', '$mark', '$date')";
     mysqli_query($connection, $sqlQuerry);
 }
 
-$sqlQuery = "SELECT idStudent FROM marks";
-$result = mysqli_query($connection, $sqlQuery);
-if (mysqli_num_rows($result) > 0) {
-    while ($row = mysqli_fetch_assoc($result)) {
-        $idStudents[] = $row["idStudent"];
-    }
-} else {
-    $error = "Rezultatų nėra";
-}
-
-for ($i = 1; $i <= $counterStudent; $i++) {
-    if (!in_array($i, $idStudents)) {
-        $sqlQuerry = "DELETE FROM students WHERE id=$i";
-        mysqli_query($connection, $sqlQuerry);
+function unusedRowDeletion($columnName, $tableName, $counterArrayName) {
+    $sqlQuery = "SELECT '$columnName' FROM marks";
+    $result = mysqli_query($GLOBALS["connection"], $sqlQuery);
```

Comment:
`$GLOBALS` masyvo naudojimas yra kind of vengtinas. Na pateisinamas tik kai kuriose situacijose. Konkrečiai šioje vietoje labai paprasta `$connection` kintamąjį perduoti kaip parametrą